gossip/store_activation_heights.go

```diff
@@ -0,0 +1,27 @@
+package gossip
+
+import (
+	"github.com/Fantom-foundation/go-opera/opera"
+)
+
+func (s *Store) AddUpgradeHeight(h opera.UpgradeHeight) {
+	orig := s.GetUpgradeHeights()
+	// allocate new memory to avoid race condition in cache
+	cp := make([]opera.UpgradeHeight, 0, len(orig)+1)
+	cp = append(append(cp, orig...), h)
+
+	s.rlp.Set(s.table.UpgradeHeights, []byte{}, cp)
+	s.cache.UpgradeHeights.Store(cp)
+}
+
+func (s *Store) GetUpgradeHeights() []opera.UpgradeHeight {
+	if v := s.cache.UpgradeHeights.Load(); v != nil {
+		return v.([]opera.UpgradeHeight)
+	}
+	hh, ok := s.rlp.Get(s.table.Blocks, []byte{}, &[]opera.UpgradeHeight{}).(*[]opera.UpgradeHeight)
```

Comment:
in `AddUpgradeHeight` you save `[]opera.UpgradeHeights` into `s.table.UpgradeHeights` table. From my perspective, you have to replace  `s.table.Blocks`  to   `s.table.UpgradeHeights` as first argument in `s.rlp.Get` method in `GetUpgradeHeights` .

Reply:
Oh by god, thank you for noticing